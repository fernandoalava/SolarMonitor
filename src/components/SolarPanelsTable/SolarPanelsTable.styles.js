import styled from "styled-components";
import Card from "@material-ui/core/Card";

export const StyledCard = styled(Card)`
  minwidth: 100;
  height: 100;
`;
