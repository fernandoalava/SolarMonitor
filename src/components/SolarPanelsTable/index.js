import React from "react";
import PropTypes from "prop-types";

import Typography from "@material-ui/core/Typography";
import Grid from "@material-ui/core/Grid";
import { StyledCard } from "./SolarPanelsTable.styles";
import CardContent from "@material-ui/core/CardContent";

export const SolarPanelsTable = props => {
  const { data } = props;
  return data.map(row => (
    <Grid key={row.id} item>
      <StyledCard>
        <CardContent>
          <Typography variant="h6" component="h3">
            {row.id}
          </Typography>
          <Typography color="textSecondary">
            Voltage: {row.voltage.toFixed(2)}
          </Typography>
          <Typography color="textSecondary">
            Wattage: {row.wattage.toFixed(2)}
          </Typography>
        </CardContent>
      </StyledCard>
    </Grid>
  ));
};

SolarPanelsTable.propTypes = {
  data: PropTypes.array.isRequired
};

SolarPanelsTable.defaultProps = {
  data: []
};

export default SolarPanelsTable;
