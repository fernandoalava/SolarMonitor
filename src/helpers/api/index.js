import config from "../../config";
import axios from "axios";

export const fetchCloudData = async (dataHandler, errorHanlder) => {
  try {
    let response = await axios.get(
      `${config.protocol}://${
        config.planetOSDomain
      }/v1/datasets/bom_access-g_global_40km/point?origin=dataset-details&apikey=${
        config.apiKey
      }&count=50&lat=49.6&lon=-50.5&var=av_ttl_cld`
    );
    dataHandler(response.data.entries);
  } catch (err) {
    errorHanlder(err);
  }
};

export const fetchSolarData = async (dataHandler, errorHanlder) => {
  try {
    let response = await axios.get(
      `${config.protocol}://${
        config.planetOSDomain
      }/v1/datasets/bom_access-g_global_40km/point?origin=dataset-details&apikey=${
        config.apiKey
      }&count=50&lat=49.6&lon=-50.5&var=av_swsfcdown`
    );
    dataHandler(response.data.entries);
  } catch (err) {
    errorHanlder(err);
  }
};
