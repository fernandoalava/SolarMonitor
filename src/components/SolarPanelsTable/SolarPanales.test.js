import { shallow } from "enzyme";
import * as React from "react";
import SolarPanelsTable from ".";
import { mockPanelsData } from "../../helpers";

describe("SolarPanelsTable", () => {
  it("Renders correctly without data", () => {
    const wrapper = shallow(<SolarPanelsTable data={[]} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("Renders correctly with data", () => {
    const wrapper = shallow(<SolarPanelsTable data={mockPanelsData} />);
    expect(wrapper).toMatchSnapshot();
  });
});
