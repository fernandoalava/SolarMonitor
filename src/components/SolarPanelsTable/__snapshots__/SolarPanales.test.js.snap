// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SolarPanelsTable Renders correctly with data 1`] = `
Array [
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <Styled(WithStyles(ForwardRef(Card)))>
      <WithStyles(ForwardRef(CardContent))>
        <WithStyles(ForwardRef(Typography))
          component="h3"
          variant="h6"
        >
          1
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          color="textSecondary"
        >
          Voltage: 
          151.02
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          color="textSecondary"
        >
          Wattage: 
          409.74
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(CardContent))>
    </Styled(WithStyles(ForwardRef(Card)))>
  </WithStyles(ForwardRef(Grid))>,
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <Styled(WithStyles(ForwardRef(Card)))>
      <WithStyles(ForwardRef(CardContent))>
        <WithStyles(ForwardRef(Typography))
          component="h3"
          variant="h6"
        >
          2
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          color="textSecondary"
        >
          Voltage: 
          154.67
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          color="textSecondary"
        >
          Wattage: 
          975.93
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(CardContent))>
    </Styled(WithStyles(ForwardRef(Card)))>
  </WithStyles(ForwardRef(Grid))>,
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <Styled(WithStyles(ForwardRef(Card)))>
      <WithStyles(ForwardRef(CardContent))>
        <WithStyles(ForwardRef(Typography))
          component="h3"
          variant="h6"
        >
          3
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          color="textSecondary"
        >
          Voltage: 
          139.49
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          color="textSecondary"
        >
          Wattage: 
          351.00
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(CardContent))>
    </Styled(WithStyles(ForwardRef(Card)))>
  </WithStyles(ForwardRef(Grid))>,
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <Styled(WithStyles(ForwardRef(Card)))>
      <WithStyles(ForwardRef(CardContent))>
        <WithStyles(ForwardRef(Typography))
          component="h3"
          variant="h6"
        >
          4
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          color="textSecondary"
        >
          Voltage: 
          118.52
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          color="textSecondary"
        >
          Wattage: 
          515.44
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(CardContent))>
    </Styled(WithStyles(ForwardRef(Card)))>
  </WithStyles(ForwardRef(Grid))>,
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <Styled(WithStyles(ForwardRef(Card)))>
      <WithStyles(ForwardRef(CardContent))>
        <WithStyles(ForwardRef(Typography))
          component="h3"
          variant="h6"
        >
          5
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          color="textSecondary"
        >
          Voltage: 
          175.13
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          color="textSecondary"
        >
          Wattage: 
          48.84
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(CardContent))>
    </Styled(WithStyles(ForwardRef(Card)))>
  </WithStyles(ForwardRef(Grid))>,
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <Styled(WithStyles(ForwardRef(Card)))>
      <WithStyles(ForwardRef(CardContent))>
        <WithStyles(ForwardRef(Typography))
          component="h3"
          variant="h6"
        >
          6
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          color="textSecondary"
        >
          Voltage: 
          160.22
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          color="textSecondary"
        >
          Wattage: 
          200.15
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(CardContent))>
    </Styled(WithStyles(ForwardRef(Card)))>
  </WithStyles(ForwardRef(Grid))>,
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <Styled(WithStyles(ForwardRef(Card)))>
      <WithStyles(ForwardRef(CardContent))>
        <WithStyles(ForwardRef(Typography))
          component="h3"
          variant="h6"
        >
          7
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          color="textSecondary"
        >
          Voltage: 
          161.19
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          color="textSecondary"
        >
          Wattage: 
          32.58
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(CardContent))>
    </Styled(WithStyles(ForwardRef(Card)))>
  </WithStyles(ForwardRef(Grid))>,
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <Styled(WithStyles(ForwardRef(Card)))>
      <WithStyles(ForwardRef(CardContent))>
        <WithStyles(ForwardRef(Typography))
          component="h3"
          variant="h6"
        >
          8
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          color="textSecondary"
        >
          Voltage: 
          164.88
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          color="textSecondary"
        >
          Wattage: 
          182.96
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(CardContent))>
    </Styled(WithStyles(ForwardRef(Card)))>
  </WithStyles(ForwardRef(Grid))>,
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <Styled(WithStyles(ForwardRef(Card)))>
      <WithStyles(ForwardRef(CardContent))>
        <WithStyles(ForwardRef(Typography))
          component="h3"
          variant="h6"
        >
          9
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          color="textSecondary"
        >
          Voltage: 
          141.20
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          color="textSecondary"
        >
          Wattage: 
          259.74
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(CardContent))>
    </Styled(WithStyles(ForwardRef(Card)))>
  </WithStyles(ForwardRef(Grid))>,
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <Styled(WithStyles(ForwardRef(Card)))>
      <WithStyles(ForwardRef(CardContent))>
        <WithStyles(ForwardRef(Typography))
          component="h3"
          variant="h6"
        >
          10
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          color="textSecondary"
        >
          Voltage: 
          120.45
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          color="textSecondary"
        >
          Wattage: 
          281.29
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(CardContent))>
    </Styled(WithStyles(ForwardRef(Card)))>
  </WithStyles(ForwardRef(Grid))>,
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <Styled(WithStyles(ForwardRef(Card)))>
      <WithStyles(ForwardRef(CardContent))>
        <WithStyles(ForwardRef(Typography))
          component="h3"
          variant="h6"
        >
          11
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          color="textSecondary"
        >
          Voltage: 
          219.77
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          color="textSecondary"
        >
          Wattage: 
          809.40
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(CardContent))>
    </Styled(WithStyles(ForwardRef(Card)))>
  </WithStyles(ForwardRef(Grid))>,
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <Styled(WithStyles(ForwardRef(Card)))>
      <WithStyles(ForwardRef(CardContent))>
        <WithStyles(ForwardRef(Typography))
          component="h3"
          variant="h6"
        >
          12
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          color="textSecondary"
        >
          Voltage: 
          136.18
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          color="textSecondary"
        >
          Wattage: 
          548.60
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(CardContent))>
    </Styled(WithStyles(ForwardRef(Card)))>
  </WithStyles(ForwardRef(Grid))>,
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <Styled(WithStyles(ForwardRef(Card)))>
      <WithStyles(ForwardRef(CardContent))>
        <WithStyles(ForwardRef(Typography))
          component="h3"
          variant="h6"
        >
          13
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          color="textSecondary"
        >
          Voltage: 
          152.19
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          color="textSecondary"
        >
          Wattage: 
          87.97
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(CardContent))>
    </Styled(WithStyles(ForwardRef(Card)))>
  </WithStyles(ForwardRef(Grid))>,
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <Styled(WithStyles(ForwardRef(Card)))>
      <WithStyles(ForwardRef(CardContent))>
        <WithStyles(ForwardRef(Typography))
          component="h3"
          variant="h6"
        >
          14
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          color="textSecondary"
        >
          Voltage: 
          197.64
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          color="textSecondary"
        >
          Wattage: 
          795.98
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(CardContent))>
    </Styled(WithStyles(ForwardRef(Card)))>
  </WithStyles(ForwardRef(Grid))>,
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <Styled(WithStyles(ForwardRef(Card)))>
      <WithStyles(ForwardRef(CardContent))>
        <WithStyles(ForwardRef(Typography))
          component="h3"
          variant="h6"
        >
          15
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          color="textSecondary"
        >
          Voltage: 
          135.09
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          color="textSecondary"
        >
          Wattage: 
          309.82
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(CardContent))>
    </Styled(WithStyles(ForwardRef(Card)))>
  </WithStyles(ForwardRef(Grid))>,
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <Styled(WithStyles(ForwardRef(Card)))>
      <WithStyles(ForwardRef(CardContent))>
        <WithStyles(ForwardRef(Typography))
          component="h3"
          variant="h6"
        >
          16
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          color="textSecondary"
        >
          Voltage: 
          195.45
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          color="textSecondary"
        >
          Wattage: 
          553.66
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(CardContent))>
    </Styled(WithStyles(ForwardRef(Card)))>
  </WithStyles(ForwardRef(Grid))>,
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <Styled(WithStyles(ForwardRef(Card)))>
      <WithStyles(ForwardRef(CardContent))>
        <WithStyles(ForwardRef(Typography))
          component="h3"
          variant="h6"
        >
          17
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          color="textSecondary"
        >
          Voltage: 
          128.71
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          color="textSecondary"
        >
          Wattage: 
          192.84
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(CardContent))>
    </Styled(WithStyles(ForwardRef(Card)))>
  </WithStyles(ForwardRef(Grid))>,
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <Styled(WithStyles(ForwardRef(Card)))>
      <WithStyles(ForwardRef(CardContent))>
        <WithStyles(ForwardRef(Typography))
          component="h3"
          variant="h6"
        >
          18
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          color="textSecondary"
        >
          Voltage: 
          142.06
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          color="textSecondary"
        >
          Wattage: 
          190.18
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(CardContent))>
    </Styled(WithStyles(ForwardRef(Card)))>
  </WithStyles(ForwardRef(Grid))>,
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <Styled(WithStyles(ForwardRef(Card)))>
      <WithStyles(ForwardRef(CardContent))>
        <WithStyles(ForwardRef(Typography))
          component="h3"
          variant="h6"
        >
          19
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          color="textSecondary"
        >
          Voltage: 
          110.59
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          color="textSecondary"
        >
          Wattage: 
          174.79
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(CardContent))>
    </Styled(WithStyles(ForwardRef(Card)))>
  </WithStyles(ForwardRef(Grid))>,
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <Styled(WithStyles(ForwardRef(Card)))>
      <WithStyles(ForwardRef(CardContent))>
        <WithStyles(ForwardRef(Typography))
          component="h3"
          variant="h6"
        >
          20
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          color="textSecondary"
        >
          Voltage: 
          171.75
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          color="textSecondary"
        >
          Wattage: 
          332.81
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(CardContent))>
    </Styled(WithStyles(ForwardRef(Card)))>
  </WithStyles(ForwardRef(Grid))>,
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <Styled(WithStyles(ForwardRef(Card)))>
      <WithStyles(ForwardRef(CardContent))>
        <WithStyles(ForwardRef(Typography))
          component="h3"
          variant="h6"
        >
          21
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          color="textSecondary"
        >
          Voltage: 
          162.89
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          color="textSecondary"
        >
          Wattage: 
          538.00
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(CardContent))>
    </Styled(WithStyles(ForwardRef(Card)))>
  </WithStyles(ForwardRef(Grid))>,
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <Styled(WithStyles(ForwardRef(Card)))>
      <WithStyles(ForwardRef(CardContent))>
        <WithStyles(ForwardRef(Typography))
          component="h3"
          variant="h6"
        >
          22
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          color="textSecondary"
        >
          Voltage: 
          116.66
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          color="textSecondary"
        >
          Wattage: 
          17.78
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(CardContent))>
    </Styled(WithStyles(ForwardRef(Card)))>
  </WithStyles(ForwardRef(Grid))>,
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <Styled(WithStyles(ForwardRef(Card)))>
      <WithStyles(ForwardRef(CardContent))>
        <WithStyles(ForwardRef(Typography))
          component="h3"
          variant="h6"
        >
          23
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          color="textSecondary"
        >
          Voltage: 
          145.72
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          color="textSecondary"
        >
          Wattage: 
          435.38
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(CardContent))>
    </Styled(WithStyles(ForwardRef(Card)))>
  </WithStyles(ForwardRef(Grid))>,
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <Styled(WithStyles(ForwardRef(Card)))>
      <WithStyles(ForwardRef(CardContent))>
        <WithStyles(ForwardRef(Typography))
          component="h3"
          variant="h6"
        >
          24
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          color="textSecondary"
        >
          Voltage: 
          111.14
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          color="textSecondary"
        >
          Wattage: 
          548.91
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(CardContent))>
    </Styled(WithStyles(ForwardRef(Card)))>
  </WithStyles(ForwardRef(Grid))>,
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <Styled(WithStyles(ForwardRef(Card)))>
      <WithStyles(ForwardRef(CardContent))>
        <WithStyles(ForwardRef(Typography))
          component="h3"
          variant="h6"
        >
          25
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          color="textSecondary"
        >
          Voltage: 
          179.49
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          color="textSecondary"
        >
          Wattage: 
          183.56
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(CardContent))>
    </Styled(WithStyles(ForwardRef(Card)))>
  </WithStyles(ForwardRef(Grid))>,
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <Styled(WithStyles(ForwardRef(Card)))>
      <WithStyles(ForwardRef(CardContent))>
        <WithStyles(ForwardRef(Typography))
          component="h3"
          variant="h6"
        >
          26
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          color="textSecondary"
        >
          Voltage: 
          201.57
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          color="textSecondary"
        >
          Wattage: 
          478.87
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(CardContent))>
    </Styled(WithStyles(ForwardRef(Card)))>
  </WithStyles(ForwardRef(Grid))>,
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <Styled(WithStyles(ForwardRef(Card)))>
      <WithStyles(ForwardRef(CardContent))>
        <WithStyles(ForwardRef(Typography))
          component="h3"
          variant="h6"
        >
          27
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          color="textSecondary"
        >
          Voltage: 
          132.30
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          color="textSecondary"
        >
          Wattage: 
          897.46
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(CardContent))>
    </Styled(WithStyles(ForwardRef(Card)))>
  </WithStyles(ForwardRef(Grid))>,
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <Styled(WithStyles(ForwardRef(Card)))>
      <WithStyles(ForwardRef(CardContent))>
        <WithStyles(ForwardRef(Typography))
          component="h3"
          variant="h6"
        >
          28
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          color="textSecondary"
        >
          Voltage: 
          144.02
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          color="textSecondary"
        >
          Wattage: 
          705.56
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(CardContent))>
    </Styled(WithStyles(ForwardRef(Card)))>
  </WithStyles(ForwardRef(Grid))>,
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <Styled(WithStyles(ForwardRef(Card)))>
      <WithStyles(ForwardRef(CardContent))>
        <WithStyles(ForwardRef(Typography))
          component="h3"
          variant="h6"
        >
          29
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          color="textSecondary"
        >
          Voltage: 
          158.34
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          color="textSecondary"
        >
          Wattage: 
          781.40
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(CardContent))>
    </Styled(WithStyles(ForwardRef(Card)))>
  </WithStyles(ForwardRef(Grid))>,
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <Styled(WithStyles(ForwardRef(Card)))>
      <WithStyles(ForwardRef(CardContent))>
        <WithStyles(ForwardRef(Typography))
          component="h3"
          variant="h6"
        >
          30
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          color="textSecondary"
        >
          Voltage: 
          150.68
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          color="textSecondary"
        >
          Wattage: 
          135.95
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(CardContent))>
    </Styled(WithStyles(ForwardRef(Card)))>
  </WithStyles(ForwardRef(Grid))>,
]
`;

exports[`SolarPanelsTable Renders correctly without data 1`] = `null`;
